src/archive_librarian/dbHandler.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * DBHandler manages an SQLite database that stores job history.
+ *
+ * Major components:
+ * - jobIdCache: avoids repeated DB hits for JobId/JobListId lookups
+ * - insertUnseenJob: populates missing User, JobList, and Job entries
+ * - batchInsertJobRecords / batchInsertEpochRecords: main entrypoints for history processing
+ * - writeFileInfo / updateFileInfo: track file-level processing state
+ * - Status tracking: records how much of each file has been processed
+ */
+
+#pragma once
+
+#include <sqlite3.h>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <list>
+#include "JobRecord.h"
+#include "archiveMonitor.h"
+#include "cache.hpp"
+
+// Hash definition so we can use it later
+struct pair_hash {
+    std::size_t operator()(const std::pair<int, int>& p) const {
+        return std::hash<int>{}(p.first) ^ (std::hash<int>{}(p.second) << 1);
+    }
+};
+
+class JobIdLookup;
+
+class DBHandler {
+public:
+    explicit DBHandler(const std::string& schemaPath, const std::string& dbPath, size_t maxCacheSize = 10000);
+    ~DBHandler();
```

Comment:
We should do rule of 5 here
```suggestion
     DBHandler() = delete;
    explicit DBHandler(const std::string& schemaPath, const std::string& dbPath, size_t maxCacheSize = 10000);

    DBHandler(const DBHandler&) = delete;
    DBHandler(DBHandler&&) = delete;
    DBHandler& operator=(const DBHandler&) = delete;
    DBHandler& operator=(DBHandler&&) = delete;

    ~DBHandler();
```